kevm-pyk/src/kevm_pyk/kproj/evm-semantics/lemmas/evm-int-simplification.k

```diff
@@ -53,6 +53,29 @@ module EVM-INT-SIMPLIFICATION-COMMON
       requires 0 <Int Y
       [simplification, comm, preserves-definedness]
 
+  // ###########################################################################
+  // #asWord
+  // ###########################################################################
+
+  rule [asWord-lt]:
+    #asWord ( BA ) <Int X => true
+    requires 2 ^Int ( lengthBytes(BA) *Int 8) <=Int X
+    [concrete(X), simplification]
+
+  rule [asWord-lt-range]:
+    #asWord ( #range ( _:Bytes, S, W ) ) <Int X => true
+    requires 0 <=Int S andBool 0 <=Int W
+     andBool 2 ^Int ( 8 *Int W ) <=Int X
+     andBool X ==Int 2 ^Int log2Int( X )
+    [simplification, concrete(S, W, X)]
```

Comment:
Just double-checking my understanding. Is the last condition `X ==Int 2 ^Int log2Int(X)` necessary? It seems like the lemma still holds without it...

Reply:
Yes, that condition is not necessary, it was just the use case in which the lemma appeared. I have removed it in #2288.

Reply:
Ok, thanks for replying! No need to remove it if it makes sense as a restriction given our cases, I just want us to be in the habit of always responding to code review questions.